### PR TITLE
Scality Isn't Quite S3

### PR DESCRIPTION
**Pull request by thomasmmitchell**
You may have thought it was S3, but it wasn't, and you were wrong.

There is now a plugin specifically for Scality. It uses a fork of the minio-go library that has the added feature of being able to set a max part size for multipart object "put"s. The Scality plugin always uses the V2 auth signature, and sets the max part size to 75MB, which is confirmed to work with Scality.

Otherwise, its pretty much the S3 plugin without the unnecessary parts. I would know - I copied that file.

### CODE REVIEW

**Comment by thomasmmitchell** [comment]
Clearly a vendoring issue - looking into it

File: plugin/scality/plugin.go [hunk 1]
```diff
@@ -0,0 +1,335 @@
...
+import (
+	"crypto/tls"
+	"fmt"
+	"io"
+	"net/http"
+	"os"
+	"time"
+
+	"github.com/jhunt/ansi"
+	minio "github.com/starkandwayne/minio-go"
```
Comment (geofffranks) [inline plugin/scality/plugin.go]: Maybe add a note that we forked minio-go, and why we need our forked version?

File: plugin/scality/plugin.go [hunk 2]
```diff
@@ -0,0 +1,335 @@
...
+	client, err := scal.Connect()
+	if err != nil {
+		return "", err
+	}
+
+	path := scal.genBackupPath()
+	plugin.DEBUG("Storing data in %s", path)
+
+	minio.SetMaxPartSize(1024 * 1024 * 75)
+	// FIXME: should we do something with the size of the write performed?
```
Comment (geofffranks) [inline plugin/scality/plugin.go]: Should this FIXME get fixed?

File: plugin/scality/plugin.go (hunk 1 above)
Comment (thomasmmitchell) [inline plugin/scality/plugin.go]: The blurb at the top of the file sort of explains that

File: plugin/scality/plugin.go (hunk 2 above)
Comment (thomasmmitchell) [inline plugin/scality/plugin.go]: Seems incorrectly marked as a FIXME. Nothing is broken. More of a... TODO? Either way, maybe print it to output? What do you think?

Comment (thomasmmitchell) [inline plugin/scality/plugin.go]: Whatever we decide, I'll mirror it in the S3 plugin

File: plugin/scality/plugin.go (hunk 1 above)
Comment (thomasmmitchell) [inline plugin/scality/plugin.go]: I'll make a specific note at the import line.

**Pull request merged**